lib/parsers/ts-utils.js

```diff
@@ -0,0 +1,54 @@
+const fs = require('node:fs');
+const ts = require('typescript');
+const { transformForLint } = require('./transform');
+const babel = require('@babel/core');
+const { replaceRange } = require('./transform');
+
+module.exports.patchTs = function patchTs() {
+  const readDirectory = ts.sys.readDirectory;
+  ts.sys.readDirectory = function (...args) {
+    const results = readDirectory.call(this, ...args);
+    return results.map((f) => f.replace(/\.gts$/, '.ts'));
+  };
+  ts.sys.fileExists = function (fileName) {
+    return fs.existsSync(fileName.replace(/\.mts$/, '.gts')) || fs.existsSync(fileName);
+  };
+  ts.sys.readFile = function (fname) {
+    let fileName = fname;
+    let content = '';
+    try {
+      content = fs.readFileSync(fileName).toString();
+    } catch {
+      fileName = fileName.replace(/\.mts$/, '.gts');
```

Comment:
```suggestion
      fileName = fileName.replace(/\.m?ts$/, '.gts');
```
I think we need to account for `.ts` extensions also. Found this is possible

Reply:
Do you know under what conditions that happens?
It should have converted all gts extensions to .mts and not .ts

Reply:
@patricklx don't we get fileNames as a result from here
https://github.com/ember-cli/eslint-plugin-ember/pull/1996/files#diff-e244765e7358fb9a23e94c9b7d23917d4ed0172bb7550fbc1ff88359691859feR11
?
It definitely did happen to me, I will try to come up with some reproduction tests tomorrow

Reply:
but basically it happend when I first create a TS program and then pass it to `gjs-gts-parser` as an argument.
For more context I'm trying to use `gjs-gts-parser` to add `gjs/gts` support to [SonarJS](https://github.com/SonarSource/SonarJS), there they create their own program [here](https://github.com/SonarSource/SonarJS/blob/master/packages/jsts/src/program/program.ts#L156) and when that happanes some of these read methos get called and I noticed error due to this.

Reply:
Ah, that should be using mts as well

Reply:
Basically what I think is happening `readDirectory` converts the files names from `gts` to `ts`, and that is how we end up with `ts` extension in fileNames inside `readFile`.

Reply:
@patricklx oh I see so you need to change
```
return results.map((f) => f.replace(/\.gts$/, '.ts'));
````
to
```
return results.map((f) => f.replace(/\.gts$/, '.gts'));
```
right?
That should also do the trick

Reply:
@patricklx thank you for fixing this whole issue so fast!
Thanks to this I was able to get SonarJS working with these latest changes, so after next alpha release I will make a PR to Sonar and see if they are willing to merge it!
But couldn't have ever done it w/o `gjs-gts-parser`!